GiniVision/Classes/Core/Helpers/Localization/LocalizableStringResource.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  Localization.swift
+//  GiniVision
+//
+//  Created by Gini GmbH on 7/31/18.
+//
+
+import Foundation
+import UIKit
+
+typealias LocalizationEntry = (value: String, description: String)
+
+protocol LocalizableStringResource {
+    var tableName: String { get }
+    var tableEntry: LocalizationEntry { get }
+    var customizable: Bool { get }
+    var args: CVarArg? { get }
+
+}
+
+extension LocalizableStringResource {
+    var localized: String {
+        let key = "ginivision.\(tableName).\(tableEntry.value)"
```

Comment:
I could argue that even though you are now using enums, the whole localization is still not type safe. In this line, you are still relying on constructing strings so they match a text file. If you typed the table name or entry wrong, the compiler wouldn't complain, but it would rather fail at runtime.

Reply:
But do you know a better way for making it safer? Because I'm not sure that I can find a better and non-complex approach for that. I know that there safer approaches like `R.swift` library, but in my opinion that's an overkill.